partiql-parser/src/main/antlr/PartiQL.g4

```diff
@@ -136,6 +137,19 @@ uniqueConstraintDef
 // but we at least can eliminate SFW query here.
 searchCondition : exprOr;
 
+// SQL Extension, Support additional table metadatas such as partition by, tblProperties, etc.
+tableExtension
+    : PARTITION BY partitionExpr                                                           # PartitionBy
+    | TBLPROPERTIES PAREN_LEFT keyValuePair (COMMA keyValuePair)* PAREN_RIGHT              # TblProperties
+    ;
+
+keyValuePair : key=LITERAL_STRING EQ value=literal;
+
+// For now: just support a list of column name
+// In the future, we might support common partition expression such as Hash(), Range(), etc.
+partitionExpr
+    : PAREN_LEFT columnName (COMMA columnName)* PAREN_RIGHT                   #PartitionColList
```

Comment:
Based on your above comment, I'd assume the other parts of HIVE DDL for `PARTITION BY`/`PARTITIONED BY` are out of scope

> `  [PARTITIONED BY (col_name data_type [COMMENT col_comment], ...)]`

namely the requirement to specify a data type along with an optional comment

Reply:
I would treat `PARTITION BY` and `PARTITIONED BY` as two different clauses.